src/main/java/edu/metrostate/ics499/team2/services/CompoundService.java

```diff
@@ -1,57 +1,75 @@
-	
-	@Autowired
-	private CompoundRepository compoundRepo;
-	
-	/**
-	 * TODO:
-	 * - Add handler for when request to PUG API doesn't return element
-	 * - Add local variable for last request sent (6 second delay if within window of last request)
-	 * - 
-	 * 
-	 */
-	
-	public boolean doesValueExistInRepo(String formula) {
-		return compoundRepo.findCompoundByFormula(formula).size() > 0;
-	}
-	
-	private Compound retrieveCompoundFromRepo(String formula) {
-		LOG.info("Value " + formula + " exists in repo, retrieving...");
-		
-		return compoundRepo.findCompoundByFormula(formula).get(0);
-	}
-	
-	private Compound retrieveCompoundFromPugApi(String formula, Compound compound) {
-		LOG.info("calling PUG API with argument: " + formula);
-		
-		PugApiDTO pugApiValue = restTemplate.getForObject(PUG_PROLOG + PUG_INPUT + formula + PUG_OPERATION + PUG_OUTPUT, PugApiDTO.class);
-		compound.setTitle(pugApiValue.getFirstPropertyTitle());
-		
-		return compoundRepo.save(compound);
-	}
-	
-	public Compound validateInput(Compound compound) {
-		String formula = compound.getFormula();	
-		
-		return doesValueExistInRepo(formula) ? retrieveCompoundFromRepo(formula) : retrieveCompoundFromPugApi(formula, compound);
-	}
-}
\ No newline at end of file
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private CompoundRepository compoundRepo;
+
+    @Autowired
+    private QuizService quizService;
+
+
+    /**
+     * TODO:
+     * - Add handler for when request to PUG API doesn't return element
+     * - Add local variable for last request sent (6 second delay if within window of last request)
+     * -
+     */
+
+    public boolean doesValueExistInRepo(String formula) {
+        return compoundRepo.findCompoundByFormula(formula).size() > 0;
+    }
+
+    private Compound retrieveCompoundFromRepo(String formula) {
+        LOG.info("Value " + formula + " exists in repo, retrieving...");
+        return compoundRepo.findCompoundByFormula(formula).get(0);
+    }
+
+    private Compound retrieveCompoundFromPugApi(String formula, Compound compound) throws PugApiException {
+        LOG.info("Calling PUG API with argument: " + formula);
+        String pubChemUrl = PUG_PROLOG + PUG_INPUT + formula + PUG_OPERATION + PUG_OUTPUT;
+        try {
+//            LOG.info("Sending PugAPI url in service: {}", pubChemUrl);
+            PugApiDTO pugApiValue = restTemplate.getForObject(pubChemUrl, PugApiDTO.class);
+            assert pugApiValue != null;
+            compound.setTitle(pugApiValue.getFirstPropertyTitle());
+            if (compound.getUserId() != null) {
+                quizService.createCompoundQuizes(compound);
+                return compoundRepo.save(compound);
+            } else
+                return compound;
+        } catch (HttpStatusCodeException exception) {
+            // to do: parse the JSON returned as an error to get a useful response
```

Comment:
I spent probably 3 hours trying to get useful JSON from the exception.

- Instead, either the controller or the call of the pugApi maybe, could return a `ResponseEntity`.

Reply:
The actual call to PUG can be handled in a separate method in order to capture the status code. The thing here is that we actually need to pass back a response object to the user so that the discovered compound can be displayed. Will add that as a TODO before the project is finalized.